e2e/dependencies/dependency1/src/lib.rs

```diff
@@ -0,0 +1,6 @@
+use ts_rs::TS;
+
+#[derive(TS)]
+pub struct LibraryType {
+    pub a: i32
+}
```

Comment:
Interesting, since this type doesn't contain `#[export]` we'd have to come up with some other way to export it in the consumer crate

Reply:
Right! I'm not sure about how to exactly solve this yet - maybe a `#[ts(export)]` type should cause all of its dependencies to be exported as well?

Reply:
We'd have to make sure it's only exported once, even if the same type is used in a bunch of places

Reply:
But this might be confusing, as the user is not explicitly `#[ts(export)]`ing the type

Reply:
> But this might be confusing, as the user is not explicitly `#[ts(export)]`ing the type

The problem in that scenario is that the user just *cant* explicitly export it, since it's in an other crate that may not be under his control.

> We'd have to make sure it's only exported once, even if the same type is used in a bunch of places

That's true, this might be the biggest challenge here.

Reply:
> The problem in that scenario is that the user just cant explicitly export it, since it's in an other crate that may not be under his control.

That is true, and it brings up the question I asked in #211:

> Should library code in `crates.io` ever implement TS? Seems like this should be an application code decision (especially the extra dependency + tests)

Reply:
(let's maybe have this discussion here then)
> Should library code in crates.io ever implement TS? Seems like this should be an application code decision (especially the extra dependency + tests)

I think it'd be a good idea to support that, yeah! These libraries would probably never `#[ts(export)]`, but making their types implement `TS` might be usefull in some cases, I think.

Reply:
> That is true, and it brings up the question I asked in https://github.com/Aleph-Alpha/ts-rs/pull/211

Speaking of which, this is getting hard to follow across two PRs, I opened a discussion (#219) to centralize this